### PR TITLE
Detect and use collocated templates

### LINKED ISSUE #55: Support for component template colocation
We currently guess the template path based on the component JS path, but with template colocation that doesn't work. We should probably try out multiple paths and use the first match that we find on the disk instead.

### PR DESCRIPTION
**Pull request by jrjohnson**
Look in multiple paths and use the first one that matches for a
template. If no paths match then output them all as an error.

Fixes #55

### CODE REVIEW

File: lib/transform.js [hunk 1]
```diff
@@ -99,17 +101,19 @@ function transform(source, template, options = {}) {
   return { source, template };
 }
 
-function guessTemplatePath(componentPath) {
+function guessPossibleTemplatePaths(componentPath) {
```
Comment (locks) [inline lib/transform.js]: What about:
```suggestion
function possibleTemplatePaths(componentPath) {
```

Comment (locks) [reply lib/transform.js]: Another concern, is the old method part of the public API of the package? I was going to suggest a deprecation path here, but the package is not 1.0 yet 🤔

Comment (Turbo87) [reply lib/transform.js]: this package does not have a public API. to only "public API" is the CLI.

Comment (jrjohnson) [reply lib/transform.js]: Done!

**Comment by Turbo87** [comment]
unfortunately it currently has conflicts due to #66.

**Comment by jrjohnson** [comment]
Rebased and squashed.

**Comment by patocallaghan** [comment]
@Turbo87 any chance of merging/releasing this?

**Comment by Turbo87** [comment]
@patocallaghan merged and released as v0.5.0. sorry for the delay 😞

**Comment by patocallaghan** [comment]
@Turbo87 amazing, thank you 🙌

**Pull request merged**